pkg/api/api.go

```diff
@@ -216,17 +214,24 @@ func Query(ctx context.Context, client promapi.Client, query string, ts time.Tim
 		q.Set("time", formatTime(ts))
 	}
 
-	_, data, warnings, err := doGetFallback(ctx, client, u, q, cache) //nolint:bodyclose
+	resp, data, warnings, err := doGetFallback(ctx, client, u, q, cache) //nolint:bodyclose
 	if err != nil {
-		return nil, warnings, err
+		var httpCode int
+		if resp != nil {
+			httpCode = resp.StatusCode
+		} else {
+			//Web Server Returned an Unknown Error.
+			httpCode = 520
+		}
+		return nil, httpCode, warnings, err
```

Comment:
I think we can simplify this like:
```suggestion
		if resp == nil {
			return nil, 520, warnings, err
		} 
		return nil, resp.StatusCode, warnings, err
```

Also I would _really_ prefer to avoid the bogus status code here - we actually do not know what happened if `resp` is nil, we do not even know if we reached the server, maybe we  failed at initiating request?

My suggestion would be to return `0` here as well and handle it accordingly in caller.

Reply:
👍